spec/classes/init_spec.rb

```diff
@@ -406,6 +407,16 @@
       end
     end
 
+    context 'with vas_conf_vasypd_update_interval set to invalid string (non-integer)' do
+      let :params do
+        { :vas_conf_vasypd_update_interval => '600invalid' }
+      end
+
+      it 'should fail' do
+        expect { should contain_class('vas') }.to raise_error(Puppet::Error, /validate_integer/)
+      end
+    end
+
```

Comment:
Please move the validation check to the 'variable type and content validations' section instead (at the bottom).
Add a new block like this to test support of integers and stringified integers:
```
      'integer' => {
        :name    => %w(vas_conf_vasypd_update_interval),
        :valid   => [242,'242'],
        :invalid => ['string', %w(array), { 'ha' => 'sh' }, 2.42, true, false, nil],
        :message => 'Expected.*to be an Integer',
      },
```

Please also add an atomic test that only tests your new parameter and nothing else. Like the test for vdi_mode for example.